servers/rendering/shader_language.cpp

```diff
@@ -9228,6 +9230,51 @@ Error ShaderLanguage::_parse_shader(const HashMap<StringName, FunctionInfo> &p_f
 				}
 #ifdef DEBUG_ENABLED
 				keyword_completion_context = CF_GLOBAL_SPACE;
+#endif // DEBUG_ENABLED
+			} break;
+			case TK_BLEND_FACTORS: {
+#ifdef DEBUG_ENABLED
+				keyword_completion_context = CF_UNSPECIFIED;
+#endif // DEBUG_ENABLED
+				shader->blend_factors.clear();
+				for (int k = 0; k < 4; k++) {
+					StringName mode;
+					_get_completable_identifier(nullptr, COMPLETION_BLEND_FACTORS, mode);
+
+					if (mode == StringName()) {
+						_set_error(RTR("Expected an identifier for blend mode."));
+						return ERR_PARSE_ERROR;
+					}
+
+					const String smode = String(mode);
+					bool found = false;
+					for (int i = 0; i < RD::BLEND_FACTOR_MAX; i++) {
+						const String name = String(RD::BLEND_FACTOR_NAMES[i]);
+						if (smode == name) {
+							found = true;
+							break;
+						}
+					}
+
+					if (!found) {
+						_set_error(vformat(RTR("Invalid blend mode: '%s'."), smode));
+						return ERR_PARSE_ERROR;
+					}
+
+					shader->blend_factors.push_back(mode);
+
+					tk = _get_token();
+					if ((k < 3) && (tk.type == TK_COMMA)) {
+						//all good, do nothing
+					} else if ((k == 3) && (tk.type == TK_SEMICOLON)) {
+						break; //done
+					} else {
+						_set_error(vformat(RTR("Unexpected token: '%s'; blend mode list must consist of exactly four mode names, separated by commas)."), get_token_text(tk)));
```

Comment:
Write numbers with the digit for better readability:

```suggestion
						_set_error(vformat(RTR("Unexpected token: '%s'; blend mode list must consist of exactly 4 mode names, separated by commas)."), get_token_text(tk)));
```